providers/implementations/kdfs/pbkdf2.c

```diff
@@ -193,6 +248,11 @@ static int kdf_pbkdf2_derive(void *vctx, unsigned char *key, size_t keylen,
         return 0;
     }
 
+#ifdef FIPS_MODULE
```

Comment:
I am a bit confused about why these checks are being done in different places... I was expecting
````
#ifdef FIPS_MODULE
  fips bounds check
#else
  existing bounds check
#endif

Reply:
In the beginning, just align it with all PRs I made: performing check in `*_derive`.

As you mentioned in other PR, if there is a better place, I should place the check there.
Therefore, moved the check to the same place now as you expected.